exporter/prometheus.go

```diff
@@ -0,0 +1,455 @@
+package exporter
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/cadvisor/info"
+	"github.com/google/cadvisor/manager"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const namespace = "container"
+
+// Just a small interface for easier housekeeping
+type vector interface {
+	Reset()
+	Describe(ch chan<- *prometheus.Desc)
+	Collect(ch chan<- prometheus.Metric)
+}
+
+type PrometheusCollector struct {
+	mutex   sync.RWMutex
+	manager manager.Manager
+
+	errors   prometheus.Counter
+	lastSeen *prometheus.CounterVec
+
+	cpuUsageSeconds       *prometheus.CounterVec
+	cpuUsageSecondsPerCPU *prometheus.CounterVec
+
+	memoryUsageBytes *prometheus.GaugeVec
+	memoryWorkingSet *prometheus.GaugeVec
+	memoryFailures   *prometheus.CounterVec
+
+	fsLimit        *prometheus.GaugeVec
+	fsUsage        *prometheus.GaugeVec
+	fsReads        *prometheus.CounterVec
+	fsReadsSectors *prometheus.CounterVec
+	fsReadsMerged  *prometheus.CounterVec
+	fsReadTime     *prometheus.CounterVec
+
+	fsWrites        *prometheus.CounterVec
+	fsWritesSectors *prometheus.CounterVec
+	fsWritesMerged  *prometheus.CounterVec
+	fsWriteTime     *prometheus.CounterVec
+
+	fsIoInProgress *prometheus.GaugeVec
+	fsIoTime       *prometheus.CounterVec
+
+	fsWeightedIoTime *prometheus.CounterVec
+
+	networkRxBytes   *prometheus.CounterVec
+	networkRxPackets *prometheus.CounterVec
+	networkRxErrors  *prometheus.CounterVec
+	networkRxDropped *prometheus.CounterVec
+	networkTxBytes   *prometheus.CounterVec
+	networkTxPackets *prometheus.CounterVec
+	networkTxErrors  *prometheus.CounterVec
+	networkTxDropped *prometheus.CounterVec
+
+	tasksSleeping        *prometheus.GaugeVec
```

Comment:
"tasks_count" broken down by type sounds good.

Although I agree it is a slippery slope. network_bytes broken down by tx, rx. Memory broken down by usage, working set, etc. Those can stay as is :)